File: apps/clinical-narrative/lib/core/models/document_models.dart
library;

class DocumentPreview {
  DocumentPreview({
    required this.html,
    required this.title,
    required this.body,
    required this.missingFields,
    required this.metadata,
  });

  final String html;
  final String title;
  final String body;
  final List<String> missingFields;
  final Map<String, dynamic> metadata;

  factory DocumentPreview.fromJson(Map<String, dynamic> json) {
    return DocumentPreview(
      html: json['html']?.toString() ?? '',
      title: json['title']?.toString() ?? '',
      body: json['body']?.toString() ?? '',
      missingFields: (json['missingFields'] as List?)
              ?.map((item) => item.toString())
              .toList() ??
          const [],
      metadata: json['metadata'] is Map<String, dynamic>
          ? Map<String, dynamic>.from(json['metadata'] as Map)
          : const {},
    );
  }
}

class DocumentRecord {
  DocumentRecord({
    required this.id,
    required this.sessionId,
    required this.documentType,
    required this.title,
    required this.body,
    required this.html,
    required this.status,
    required this.createdAt,
    required this.updatedAt,
    required this.metadata,
  });

  final String id;
  final String sessionId;
  final String documentType;
  final String title;
  final String body;
  final String html;
  final String status;
  final String createdAt;
  final String updatedAt;
  final Map<String, dynamic> metadata;

  factory DocumentRecord.fromJson(Map<String, dynamic> json) {
    return DocumentRecord(
      id: json['_id']?.toString() ?? '',
      sessionId: json['sessionId']?.toString() ?? '',
      documentType: json['documentType']?.toString() ?? '',
      title: json['title']?.toString() ?? '',
      body: json['body']?.toString() ?? '',
      html: json['html']?.toString() ?? '',
      status: json['status']?.toString() ?? '',
      createdAt: json['createdAt']?.toString() ?? '',
      updatedAt: json['updatedAt']?.toString() ?? '',
      metadata: json['metadata'] is Map<String, dynamic>
          ? Map<String, dynamic>.from(json['metadata'] as Map)
          : const {},
    );
  }
}

File: services/survey-backend/app/domain/models/document_models.py
from datetime import datetime
from typing import Optional, Dict, Any, List

from pydantic import BaseModel, Field, ConfigDict


class DocumentPreviewRequest(BaseModel):
    session_id: str = Field(..., alias="sessionId")
    document_type: str = Field(..., alias="documentType")
    title: Optional[str] = None
    body: Optional[str] = None

    model_config = ConfigDict(extra="forbid", populate_by_name=True)


class DocumentExportRequest(BaseModel):
    session_id: str = Field(..., alias="sessionId")
    document_type: str = Field(..., alias="documentType")
    title: str
    body: str
    metadata: Optional[Dict[str, Any]] = None

    model_config = ConfigDict(extra="forbid", populate_by_name=True)


class DocumentPreview(BaseModel):
    html: str
    title: str
    body: str
    missing_fields: List[str] = Field(default_factory=list, alias="missingFields")
    metadata: Dict[str, Any] = Field(default_factory=dict)

    model_config = ConfigDict(extra="forbid", populate_by_name=True)


class DocumentRecord(BaseModel):
    id: Optional[str] = Field(default=None, alias="_id")
    session_id: str = Field(..., alias="sessionId")
    document_type: str = Field(..., alias="documentType")
    title: str
    body: str
    html: str
    status: str = Field(default="final")
    created_at: datetime = Field(default_factory=datetime.utcnow, alias="createdAt")
    updated_at: datetime = Field(default_factory=datetime.utcnow, alias="updatedAt")
    metadata: Optional[Dict[str, Any]] = None

    model_config = ConfigDict(extra="forbid", populate_by_name=True)
